src/app/filters/examples/filter-example.component.ts

```diff
@@ -0,0 +1,319 @@
+          value: 'December'
+        }]
+      }, {
+        id: 'weekDay',
+        title: 'Week Day',
+        placeholder: 'Filter by Week Day...',
+        type: 'typeahead',
+        queries: [
+          ...this.weekDayQueries
+        ]
+      }, {
+        id: 'withimage',
+        title: 'With Image',
+        placeholder: 'Filter by Items...',
+        type: 'typeahead',
+        queries: [
+          ...this.imageQueriesFixed,
+          this.separator,
+          ...this.imageQueries
+        ]
+      }, {
+        id: 'withicon',
+        title: 'With Icon',
+        placeholder: 'Filter by Icons...',
+        type: 'select',
+        queries: [{
+          id: 'bookmark',
+          value: 'Bookmark',
+          iconClass: 'fa-bookmark'
+        }, {
+          id: 'map',
+          value: 'Map',
+          iconClass: 'fa-map-marker'
+        }, {
+          id: 'gift',
+          value: 'Gift',
+          iconClass: 'fa-gift'
+        }]
+      }] as FilterField[],
+      resultsCount: this.items.length,
+      appliedFilters: []
+    } as FilterConfig;
+  }
+
+  // Filter functions
+
+  applyFilters(filters: Filter[]): void {
+    this.items = [];
+    if (filters && filters.length > 0) {
+      this.allItems.forEach((item) => {
+        if (this.matchesFilters(item, filters)) {
+          this.items.push(item);
+        }
+      });
+    } else {
+      this.items = this.allItems;
+    }
+    this.filterConfig.resultsCount = this.items.length;
+  }
+
+  // Reset filtered queries
+  fieldSelected(event: FilterEvent): void {
+    this.filterConfig.fields.forEach((field) => {
+      if (field.id === 'withimage') {
+        field.queries = [
+          ...this.imageQueriesFixed,
+          this.separator,
+          ...this.imageQueries
+        ];
+      } else if (field.id === 'weekDay') {
+        field.queries = [
+          ...this.weekDayQueries
+        ];
+      }
+    });
+  }
+
+  filterChange(event: FilterEvent): void {
+    this.filtersText = '';
+    event.appliedFilters.forEach((filter) => {
+      this.filtersText += filter.field.title + ' : ' + filter.value + '\n';
+    });
+    this.applyFilters(event.appliedFilters);
+    this.fieldSelected(event);
+  }
+
+  matchesFilter(item: any, filter: Filter): boolean {
+    let match = true;
+    if (filter.field.id === 'name') {
+      match = item.name.match(filter.value) !== null;
+    } else if (filter.field.id === 'address') {
+      match = item.address.match(filter.value) !== null;
+    } else if (filter.field.id === 'birthMonth') {
+      match = item.birthMonth === filter.value;
+    } else if (filter.field.id === 'weekDay') {
+      match = item.weekDay === filter.value;
+    }
+    return match;
+  }
+
+  matchesFilters(item: any, filters: Filter[]): boolean {
+    let matches = true;
+    filters.forEach((filter) => {
+      if (!this.matchesFilter(item, filter)) {
+        matches = false;
+        return false;
+      }
+    });
+    return matches;
+  }
```

Comment:
Not sure I understand why 'matches' exists. Why set it to false just to return false?